chains/nomad-ethereum/src/home.rs

```diff
@@ -157,54 +160,61 @@ where
 
 /// A reference to a Home contract on some Ethereum chain
 #[derive(Debug)]
-pub struct EthereumHome<M>
+pub struct EthereumHome<W, R>
 where
-    M: ethers::providers::Middleware,
+    W: ethers::providers::Middleware + 'static,
+    R: ethers::providers::Middleware + 'static,
 {
-    contract: Arc<EthereumHomeInternal<M>>,
+    write_contract: Arc<EthereumHomeInternal<W>>,
+    read_contract: Arc<EthereumHomeInternal<R>>,
     domain: u32,
     name: String,
-    provider: Arc<M>,
 }
 
-impl<M> EthereumHome<M>
+impl<W, R> EthereumHome<W, R>
 where
-    M: ethers::providers::Middleware + 'static,
+    W: ethers::providers::Middleware + 'static,
+    R: ethers::providers::Middleware + 'static,
 {
     /// Create a reference to a Home at a specific Ethereum address on some
     /// chain
     pub fn new(
-        provider: Arc<M>,
+        write_provider: Arc<W>,
+        read_provider: Arc<R>,
         ContractLocator {
             name,
             domain,
             address,
         }: &ContractLocator,
     ) -> Self {
         Self {
-            contract: Arc::new(EthereumHomeInternal::new(
+            write_contract: Arc::new(EthereumHomeInternal::new(
                 address.as_ethereum_address().expect("!eth address"),
-                provider.clone(),
+                write_provider,
```

Comment:
use `contract.connect` to clone with a different provider

https://docs.rs/ethers/latest/ethers/contract/struct.Contract.html#method.connect

Reply:
nvm: https://github.com/gakonst/ethers-rs/issues/1158